src/python/pants/engine/intrinsics.py

```diff
@@ -120,13 +123,57 @@ async def run_id() -> RunId:
     return await native_engine.run_id()
 
 
+__SQUELCH_WARNING = "__squelch_warning"
+
+
+# NB: Call one of the helpers below, instead of calling this rule directly,
+#  to ensure correct application of restartable logic.
 @_uncacheable_rule
-async def interactive_process(
+async def _interactive_process(
     process: InteractiveProcess, process_execution_environment: ProcessExecutionEnvironment
 ) -> InteractiveProcessResult:
+    # This is a crafty way for a caller to signal into this function without a dedicated arg
+    # (which would confound the solver).  Note that we go via __dict__ instead of using
+    # setattr/delattr, because those error for frozen dataclasses.
+    if __SQUELCH_WARNING in process.__dict__:
+        del process.__dict__[__SQUELCH_WARNING]
+    else:
+        logging.warning(
```

Comment:
Should we do this as an explicit `warn_or_error` deprecation for a version or two in advance?

Reply:
Well, this isn't really a deprecation - calling the rule directly is already broken, so there is nothing we're supporting through a deprecation period, nor is there anything we'd be changing after that period is over. We can't stop people awaiting the Effect (at least until we enforce call-by-name, at which point we can get rid of the warning).